Pattern Data Mining

In [89]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [16]:
df = pd.read_csv('travel_data.csv')

In [17]:
print(df.columns)

Index(['phone_no', 'travel_company', 'full_name', 'trip_no', 'ticket_no',
       'travel_time', 'travel_type', 'office_location', 'departure', 'arrival',
       'seat_state', 'booking_time', 'seat_no', 'ticket_price',
       'payment_method', 'type_of_trave', 'total_tickets'],
      dtype='object')


In [5]:
print(df.iloc[0:4])

   phone_no  travel_company         full_name  trip_no  ticket_no  \
0   1871516             532       �MER TEKTA?    75455   11685431   
1   1847377             476  TU?RUL HAFIZO?LU   417832   15055943   
2   1845989             140      H�SEYIN KAYA    95236   11500972   
3   1853372             693      ERS?N ATMACA    26093   10590360   

        travel_time  travel_type office_location   departure    arrival  \
0  01/01/2012 00:00            0      D?YARBAKIR  D?YARBAKIR     MERS?N   
1  01/01/2012 00:00            0          SAMSUN      SAMSUN    KAYSER?   
2  01/01/2012 01:00            0        ?STANBUL    ?STANBUL  ESK??EH?R   
3  01/01/2012 01:00            0           N??DE       N??DE     ANKARA   

   seat_state booking_time  seat_no  ticket_price  payment_method  \
0           3      30:10.0       37          40.0               1   
1           3      16:50.0       27          45.0               1   
2           3      54:00.0        3          35.0               1   
3 

In [72]:
df.describe()

phone_no  travel_company       trip_no     ticket_no  \
count  1.000000e+05   100000.000000  1.000000e+05  1.000000e+05   
mean   1.857025e+06     1248.205490  1.725866e+05  1.127590e+07   
std    1.298754e+04    11889.953033  2.277385e+05  8.619154e+06   
min    1.839010e+06       14.000000  0.000000e+00  0.000000e+00   
25%    1.847156e+06      168.000000  3.490350e+04  2.086196e+06   
50%    1.853256e+06      434.000000  8.883550e+04  1.156559e+07   
75%    1.864801e+06      686.000000  1.959520e+05  1.368801e+07   
max    1.897167e+06   211400.000000  1.477461e+06  3.756544e+07   

         travel_type     seat_state        seat_no   ticket_price  \
count  100000.000000  100000.000000  100000.000000  100000.000000   
mean        0.027310       2.469120      20.453490      49.440086   
std         0.618715       0.883139      13.210527      23.063687   
min         0.000000       1.000000       1.000000       0.000000   
25%         0.000000       1.000000       9.000000      30.000000   
50%         0.000000       3.000000      19.000000      50.000000   
75%         0.000000       3.000000      31.000000      65.000000   
max        35.000000       3.000000      78.000000     220.500000   

       payment_method  total_tickets     count  
count   100000.000000  100000.000000  100000.0  
mean         0.829860      45.058810       1.0  
std          0.382327      17.430571       0.0  
min          0.000000       1.000000       1.0  
25%          1.000000      34.000000       1.0  
50%          1.000000      45.000000       1.0  
75%          1.000000      54.000000       1.0  
max          3.000000     178.000000       1.0

In [20]:
df.info

<bound method DataFrame.info of        phone_no  travel_company          full_name  trip_no  ticket_no  \
0       1871516             532        �MER TEKTA?    75455   11685431   
1       1847377             476   TU?RUL HAFIZO?LU   417832   15055943   
2       1845989             140       H�SEYIN KAYA    95236   11500972   
3       1853372             693       ERS?N ATMACA    26093   10590360   
4       1846628             140      ERCAN DEMIRCI    95056   11498257   
...         ...             ...                ...      ...        ...   
99995   1871203             987        EK?N S�NMEZ    29484     684315   
99996   1853278             434       VURAL KESK?N   227036   11915976   
99997   1859522             196         �ZCAN �ZER    18657     277884   
99998   1847781            1463  F?KRETKI?LALIOGLU    13728     144871   
99999   1847781            1463  F?KRETKI?LALIOGLU    13728     144872   

            travel_time  travel_type office_location   departure  \
0      01/0

**Trips that are purchased most frequently**

In [27]:
trip_no = df.iloc[:,3]

In [63]:
pd.unique(trip_no)

array([ 75455, 417832,  95236, ..., 227036,  18657,  13728], dtype=int64)

In [55]:
trip_no.count()

100000

In [65]:
df['trip_no'].value_counts()

6164      14
15793     14
25086     13
28582     13
9006      12
          ..
9822       1
375853     1
238638     1
873520     1
298614     1
Name: trip_no, Length: 60436, dtype: int64

The most purchased trips were trips **6164** and **15793** both purchased 14 times

**Companies that are preferred most frequently**

In [66]:
df['travel_company'].value_counts()

105     10019
476      5636
175      3022
1603     2802
308      2751
        ...  
2002        1
1813        1
1883        1
1379        1
1330        1
Name: travel_company, Length: 232, dtype: int64

The most preferred travel company is company **105**

**Trips purchased most frequently** by men and women where 3 = man, 1 = woman

In [84]:
fr_trips = df.iloc[:,[3,10]]

In [87]:
print(fr_trips)

       trip_no  seat_state
0        75455           3
1       417832           3
2        95236           3
3        26093           3
4        95056           3
...        ...         ...
99995    29484           3
99996   227036           3
99997    18657           3
99998    13728           3
99999    13728           3

[100000 rows x 2 columns]


In [101]:
df['count'] = 1


df.groupby(['trip_no', 'seat_state']).count()['count']

trip_no  seat_state
0        1             2
         3             6
3        3             1
6        1             2
         3             1
                      ..
1410216  3             2
1410343  3             1
1420493  1             2
1422951  3             1
1477461  3             1
Name: count, Length: 63872, dtype: int64